crates/qasm3/src/lib.rs

```diff
@@ -150,11 +154,147 @@ pub fn load(
     loads(py, source, custom_gates, include_path)
 }
 
+#[derive(Debug, Clone)]
+struct DumpOptions {
+    includes: Vec<String>,
+    basis_gates: Vec<String>,
+    disable_constants: bool,
+    allow_aliasing: bool,
+    indent: String,
+}
+
+impl Default for DumpOptions {
+    fn default() -> Self {
+        Self {
+            includes: vec!["stdgates.inc".to_string()],
+            basis_gates: vec![],
+            disable_constants: true,
+            allow_aliasing: false,
+            indent: "  ".to_string(),
+        }
+    }
+}
```

Comment:
If these are just the options that go into making an `Exporter`, would it have made sense to skip the intermediate structure and go straight to having constructors of `Exporter`?  `Exporter` itself is just a configuration definition for `QASM3Builder`, and this is a configuration definition for `Exporter`.